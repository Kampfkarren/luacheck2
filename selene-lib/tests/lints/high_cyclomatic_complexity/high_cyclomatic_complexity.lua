local React = {
    createElement = function(...) end
}
local function withStyle(x) return x end

return withStyle(function(props)
        return React.createElement("TextLabel", { style = if props.blue then 0 else 1 },{
            Child1 = props.mask and React.createElement("Instance") or nil,
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
            React.createElement(if _G.__PROFILING__ then "TextLabel" elseif _G.__DEV__ then "Instance" else "Non"),
        })({
            if props == nil then "mallet" else "alice"
        })
    end)(function()
        if _G.__DEV__ then
            print("howdy")
        end
    end)()
