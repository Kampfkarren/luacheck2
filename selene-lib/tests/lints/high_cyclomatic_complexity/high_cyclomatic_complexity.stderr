error[limit_function_complexity]: cyclomatic complexity is too high (55 > 40)
  ┌─ selene-lib/tests/lints/high_cyclomatic_complexity/high_cyclomatic_complexity.lua:9:1
  │
9 │ function Reconciler:beginWork(current, lockInLochness, renderBikes)
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[limit_function_complexity]: cyclomatic complexity is too high (49 > 40)
    ┌─ selene-lib/tests/lints/high_cyclomatic_complexity/high_cyclomatic_complexity.lua:241:7
    │
241 │ local function MyComponent(purse)
    │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[limit_function_complexity]: cyclomatic complexity is too high (49 > 40)
    ┌─ selene-lib/tests/lints/high_cyclomatic_complexity/high_cyclomatic_complexity.lua:381:18
    │
381 │ return withStyle(function(purse)
    │                  ^^^^^^^^^^^^^^^
